lib/handlers/allow.js

```diff
@@ -93,56 +102,69 @@ function fetchDocument (host, ldp, baseUri) {
   }
 }
 
-function getUserId (req, callback) {
-  var onBehalfOf = req.get('On-Behalf-Of')
-  if (!onBehalfOf) {
-    return callback(null, req.session.userId)
+/**
+ * Determines the id of the current user from one of the following:
+ *
+ * - From `On-Behalf-Of` header (if it's present AND secretary is authorized)
+ * - From `session.userId` (if no delegation header, or secretary unauthorized)
+ *
+ * @method getUserId
+ * @param webId {string} Current user, from session.userId
+ * @param onBehalfOf {string} Contents of the `On-Behalf-Of` delegation header
+ * @param fetchDocument {Function} Result of `documentFetcher()` call
+ * @return {Promise<string>} The user id (either the current user, or
+ *   the user being delegated for).
+ */
+function getUserId (webId, onBehalfOf, fetchDocument) {
+  let principal = onBehalfOf
+  if (!webId || !principal) {
```

Comment:
Or does that to handle unauthenticated users? If we're trying to return a null value for when the request is unauthenticated I'd prefer to explicitly return `null`.